Develop3

MaritimeAst opened the pull request:
Проект выполнен с использованием Selenide
Добавлены тесты по следующим сценариям:

-Успешная регистрация.
-Ошибка для некорректного пароля. Минимальный пароль — шесть символов.

-Вход по кнопке «Войти в аккаунт» на главной,
-Вход через кнопку «Личный кабинет»,
-Вход через кнопку в форме регистрации,
-Вход через кнопку в форме восстановления пароля.
-Выход по кнопке «Выйти» в личном кабинете

-Переход в личный кабинет
-Переход по клику на «Личный кабинет».
-Переход из личного кабинета в конструктор
-Переход по клику на «Конструктор» и на логотип Stellar Burgers.

Переходы к разделам:
-«Булки»,
-«Соусы»,
-«Начинки».

MaritimeAst committed:
Добавлены тесты на регистрацию

MaritimeAst committed:
Добавлены тесты на регистрацию

MaritimeAst committed:
Добавлены тесты на login и переходы

MaritimeAst committed:
Добавлены тесты на login и переходы

MaritimeAst committed:
Добавлены тесты на login и переходы

MaritimeAst committed:
Рефакторинг тестов и добавление allure отчета

MaritimeAst committed:
Рефакторинг тестов и добавление параметра для запуска в других браузерах. По умолчанию тесты выполняются в chrome

pomegranate282922 commented on src/main/java/models/User.java:
```diff
@@ -0,0 +1,39 @@
+public class User {
+
+
+    public String email;
```
Нужно применить принцип инкапсуляции.

pomegranate282922 commented on src/main/java/pom/MainPage.java:
```diff
@@ -0,0 +1,37 @@
+    @Step("Переход к разделу Начинки")
+    public void transitionToFilling(){
+        fillingDiv.click();
+        fillingH2.shouldBe(Condition.visible);
```
Здесь есть ошибка в логике.
ты проверяешь текст в разделах, но метод isDisplayed() (и Condition.visible) на самом деле считает, что текст виден, даже если он проскролен, т.к. этот метод проверяет что у локатора отсутствует свойство hidden.
Попробуй проскролить разделы и смотреть как меняются имена классов в верстке у разделов. И построй проверку "в каком разделе я нахожусь" на этой основе.

pomegranate282922 commented on src/test/java/LoginTest.java:
```diff
@@ -0,0 +1,108 @@
+
+        user = UserGenerator.getDefault();
+        userClient = new UserClient();
+        personalCabinetPage = open("https://stellarburgers.nomoreparties.site", PersonalCabinetPage.class);
```
Урлы выносим как константы . Так их проще переиспользовать и изменять.

pomegranate282922 commented on src/test/java/RegistrationErrorTest.java:
```diff
@@ -0,0 +1,34 @@
+        personalCabinetPage.registration(user.name, user.email, user.password);         //Заполнение формы регистрации
+        personalCabinetPage.registrationSubmit();                                       //Подтверждение регистрации
+        personalCabinetPage.registrationPasswordError();                                //Проверка появления сообщения об ошибке
+    }
```
Этот тест тоже может словить баг и создать юзера. В таком случае юзер должен быть удалён.

pomegranate282922 reviewed (commented):
Чистая работа! Хорошие локаторы, удачная структура.
Пару неточностей нужно подправить - оставила в коде комментарии.

MaritimeAst committed:
Исправление недочетов по итогам ревью
1) Исправлен модификатор доступа в UserClient
2) Исправлена логика тестов перехода к разделам конструтора. Проверка на появление класса, отвечающего за выделение выбранного таба
3) Добавлено удаление пользователя в негативном тесте регистрации пользователя с коротким паролем, в случае если появится баг в регистрации
4) URL вынессены в константы

pomegranate282922 commented on pom.xml:
```diff
@@ -8,9 +8,108 @@
     <properties>
         <maven.compiler.source>17</maven.compiler.source>
         <maven.compiler.target>17</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Java 11, да?

pomegranate282922 commented on src/main/java/clients/UserClient.java:
```diff
@@ -0,0 +1,48 @@
+    public ValidatableResponse login(String login, String password){
+        return given()
+                .spec(getSpec())
+                .body("{ \"email\": \"" +  login + "\", \"password\": \""  + password + "\"}")
```
Здесь некрасивая передача тела. В простейших случаях лучше использовать мапу.

pomegranate282922 commented on src/test/java/TransitionToIngredients.java:
```diff
@@ -0,0 +1,60 @@
+    public void goToTheSaucesTab() {
+        mainPage.clickSausesButton();
+        String actual = mainPage.getClassSaucesButton();
+        String expected = "tab_tab__1SPyG tab_tab_type_current__2BEPc pt-4 pr-10 pb-4 pl-10 noselect";
```
Вообще, лучше анализировать один конкретный класс Элемента. Так тест станет стабильнее.

pomegranate282922 reviewed (approved):
Зачёт!
